Adapter/PhpcrOdmAdapter.php

```diff
@@ -190,12 +203,42 @@ public function getReferringAutoRoutes($contentDocument)
     public function findRouteForUri($uri, UriContext $uriContext)
     {
         $path = $this->getPathFromUri($uri);
-
-        return $this->dm->find(null, $path);
+        $node = $this->dm->find(null, $path);
+        if ($node instanceof AutoRouteInterface) {
+            return $node;
+        }
+        return false;
```

Comment:
Return `null` instead of `false` I think? The document manager will return `null` if it found no document.